src/Database/DBStructure.php

```diff
@@ -755,9 +755,9 @@ public static function definition() {
 		$database["config"] = [
 				"comment" => "main configuration storage",
 				"fields" => [
-						"id" => ["type" => "int", "not null" => "1", "extra" => "auto_increment", "primary" => "1", "comment" => ""],
-						"cat" => ["type" => "varbinary(255)", "not null" => "1", "default" => "", "comment" => ""],
-						"k" => ["type" => "varbinary(255)", "not null" => "1", "default" => "", "comment" => ""],
+						"id" => ["type" => "int unsigned", "not null" => "1", "extra" => "auto_increment", "primary" => "1", "comment" => ""],
+						"cat" => ["type" => "varbinary(50)", "not null" => "1", "default" => "", "comment" => ""],
```

Comment:
The config table is really used very often. This really could increase performance since it should reduce index size and with this the memory usage and query time.

Reply:
About that, I don't think the config tables should be used that often. I have a plan for making less requests to the various config tables by loading the whole config at once at the start of the script, and writing the changed value in one request on script exit.

Reply:
We already are having some memory caching method in the config class. But there are also cases where we have to use uncached config values. Writing back only on exit could create massive problems if some process is crashing.

I guess there are config values that need to be written and read unbuffered. Surely not all, but some.

Reply:
We are indeed caching the result of multiple single-value requests, while a single catchall request would actually be more efficient although more memory intensive, but we don't have that many config values anyway.

Understood for the writing cache.

PHP allows self-contained script executions, so I don't think we need unbuffered config reads at all. Even if a config change, we can update the memory cache to avoid a re-fetch from the DB.

Reply:
There are several value caching methods in PHP. Possibly we could use one of them.